src/emsarray/conventions/_base.py

```diff
@@ -1973,7 +1973,7 @@ def hash_geometry(self, hash: "hashlib._Hash") -> None:
             # Include the dtype of the data array.
             # A float array and an int array mean very different things,
             # but could have identical byte patterns.
-            hash_string(hash, data_array.encoding['dtype'].name)
+            hash_string(hash, data_array.encoding.get('dtype', data_array.values.dtype).name)
```

Comment:
The `dtype` encoding value should be present most of the time. It being missing is an unexpected exception to the norm. Without the context of this bug checking `data_array.values.dtype` looks unnecessary.

Please add brief one or two line description of why we are doing this extra step, and include a link to the xarray bug report tracking this issue: https://github.com/pydata/xarray/issues/2436

Reply:
I've added a comment with issue link and a simple description.